stately/src/otherNativeMain/kotlin/co/touchlab/stately/concurrency/Lock.kt

```diff
@@ -1,9 +1,21 @@
 package co.touchlab.stately.concurrency
 
+import co.touchlab.stately.freeze
 import kotlin.native.concurrent.AtomicInt
+import kotlin.native.concurrent.ThreadLocal
 
+@ThreadLocal
+private object Thread {
+  val id: Int by lazy { Any().hashCode() }
+}
+
+/*
+ * This is a re-entrant implementation of Lock.
+ * a thread may acquire the lock as many times as the wish.
+ */
 actual class Lock actual constructor() {
-  val lock = AtomicInt(0)
+  private val lock = AtomicInt(0)
+  private val reenterCount = AtomicInt(0)
```

Comment:
```suggestion
  private val lockCount = AtomicInt(0)
```

Reply:
I would prefer to leave it reenterCount since we aren't adding to, or removing from this when we initially lock or unlock. It is keeping track of reenteries only.